lib/Localization/LocalizationFormat.cpp

```diff
@@ -110,9 +115,9 @@ LocalizationProducer::getMessageOr(swift::DiagID id,
   if (localizedMessage.empty())
     return defaultMessage;
   if (printDiagnosticNames) {
-    llvm::StringRef diagnosticID(DiagnosticEngine::diagnosticIDStringFor(id));
+    llvm::StringRef diagnosticName(diagnosticNameStrings[(unsigned)id]);
     auto localizedDebugDiagnosticMessage =
-        localizationSaver.save(localizedMessage.str() + " [" + diagnosticID.str() + "]");
```

Comment:
Could you please elaborate what is regressed? Valid code shouldn't be impacted by this at all and `printDiagnosticNames` is turned off by default.

Reply:
This change should be "NFC", the regression was to build time due to introducing a dependency on libAST.  See https://github.com/apple/swift/pull/37320#issuecomment-844333864

Reply:
Ah okay, sounds good then!